src/haddock/modules/topology/default.py

```diff
@@ -72,43 +104,42 @@ def run(self, module_information):
-            #  are found
-            expected = []
-            not_found = []
-            for model in models:
-                model_name = model.stem
-                processed_pdb = (self.path /
-                                 f"{model_name}_haddock.{Format.PDB}")
-                if not processed_pdb.is_file():
-                    not_found.append(processed_pdb.name)
-                processed_topology = (self.path /
-                                      f"{model_name}_haddock"
-                                      f".{Format.TOPOLOGY}")
-                if not processed_topology.is_file():
-                    not_found.append(processed_topology.name)
-                topology = TopologyFile(processed_topology,
-                                        path=(self.path / TOPOLOGY_PATH))
-                expected.append(topology)
-                expected.append(PDBFile(processed_pdb,
-                                        topology,
-                                        path=(self.path / TOPOLOGY_PATH)))
-            if not_found:
-                self.finish_with_error("Several files were not generated:"
-                                       f" {not_found}")
-
-            # Save module information
-            io = ModuleIO()
-            for model in models:
-                io.add(PDBFile(model))
-            io.add(expected, "o")
-            io.save(self.path)
+        # Run CNS engine
+        logger.info(f"Running CNS engine with {len(jobs)} jobs")
+        engine = CNSEngine(jobs)
+        engine.run()
+        logger.info("CNS engine has finished")
+
+        # Check for generated output, fail it not all expected files
+        #  are found
+        expected = []
+        not_found = []
+        for model in models:
+            model_name = model.stem
+            processed_pdb = (self.path / f"{model_name}_haddock.{Format.PDB}")
+            if not processed_pdb.is_file():
+                not_found.append(processed_pdb.name)
+            processed_topology = (self.path /
+                                  f"{model_name}_haddock"
+                                  f".{Format.TOPOLOGY}")
+            if not processed_topology.is_file():
+                not_found.append(processed_topology.name)
+            topology = TopologyFile(processed_topology,
+                                    path=(self.path / TOPOLOGY_PATH))
+            expected.append(topology)
+            expected.append(PDBFile(processed_pdb,
+                                    topology,
+                                    path=(self.path / TOPOLOGY_PATH)))
+        if not_found:
+            self.finish_with_error("Several files were not generated:"
+                                   f" {not_found}")
+
+        # Save module information
+        io = ModuleIO()
+        for model in models:
+            io.add(PDBFile(model))
+        io.add(expected, "o")
+        io.save(self.path)
```

Comment:
This block before was inside the `for` loop. Now it is outside. Is this intended?

Reply:
yes, before it was generating the topology for the models into an assembly, now it does both the models from the assembly and different inputs.

Reply:
Is is able to continue from a half-run run? If you tested it and it works, perfect.